### DISCUSSION TITLE
Applying custom Huggingface model as an encoder

### DISCUSSION THREAD

**Discussion by efima-ai** [ludwig-ai/ludwig#2057] [Q&A] [answered]
Hi,

The documentation says 

_The available encoders include encoders used for [Sequence Features](https://ludwig-ai.github.io/ludwig-docs/0.5/configuration/features/sequence_features#sequence-input-features-and-encoders) **as well as encoders from the huggingface transformers library: bert, gpt, gpt2, xlnet, xlm, roberta, distilbert, ctrl, camembert, albert, t5, xlmroberta, flaubert, electra, longformer and auto-transformer**._

Does this mean that Ludwig does not support a model that is found on Huggingface but not listed above, e.g. https://huggingface.co/TurkuNLP/bert-base-finnish-cased-v1, as an encoder for the text features?

Thanks 😊

**Comment by jppgks** [comment, marked as answer]
Hi @efima-ai, you can use arbitrary HuggingFace models using the auto_transformer encoder. You can try something like this for your case:

```python
import pandas as pd
import yaml

from ludwig.api import LudwigModel

config = """
input_features:
    - name: text
      type: text
      encoder: auto_transformer
      pretrained_model_name_or_path: 'TurkuNLP/bert-base-finnish-cased-v1'
output_features:
    - name: category
      type: category
trainer:
    epochs: 1
"""
model = LudwigModel(yaml.load(config), backend="local")

df = pd.DataFrame(
    {
        "text": ["Suomessa vaihtuu kesän aikana sekä pääministeri että valtiovarain"],
        "category": ["Suomi"],
    }
)
model.train(df)
model.predict(df)
```

**Reply by efima-ai** [reply]
Thanks Joppe. Finally got time to try this out - and it works fantastically!
